samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -107,13 +127,24 @@ def __init__(
         self._aws_region = aws_region
         self._aws_profile = aws_profile
 
+        # set the default container mode
+        if not container_mode:
+            container_mode = ContainerMode.COLD.value
```

Comment:
why do we have `.value` here, isn't `container_mode` an Enum type?

Reply:
`container _mode` is string, I do not think click can define enum options